### DISCUSSION TITLE
Encourage users to submit their patterns to the pattern directory

### DISCUSSION THREAD

**Discussion by carolinan** [WordPress/gutenberg#52962] [General]
Now that users can create their own patterns more easily we could:

- Alert users that they can submit patterns by adding a notification in the pattern editing interface. Include links to more information, requirements etc.
- Long term, make it possible to submit a pattern to the directory directly from the pattern editing interface.

What do you think? @kafleg @annezazu @ryelle

**Comment by annezazu** [comment]
I think that would be really neat with very clear guardrails! My biggest concern is that there are restrictions for the Pattern Directory that might be hard to work around or explain visually. We also want to optimize the directory for high quality patterns rather than accidental submissions. I do think, in the same way we could think about submitting photos from one's Media Library to WordPress photos, we should have a shared pathway for this though. A "contribution mode" or something :)

**Comment by carolinan** [comment]
I understand why we say that we want high quality patterns. But I also feel strongly that WordPress needs to continue to be open to all contributors. Wether they are a great designer or not.

**Comment by Ren2049** [comment]
The pattern directory would be more useful if it had some kind of legend thumbnail previews of pattern categories. It improves decision making because the're more nuances to patterns than names that fit within interfaces can't express without cluttering the interface.

E.g. a hero pattern can take many shapes and forms, and currently users have to sift through a ton of irrelevant hero patterns until they find what they want (e.g. a two column hero with image on the left and text on the right). This would be much easier to express in a thumbnail. Once they click on a thumbnail it then would reveal user submitted two column patterns associated with it.

Uploaders would have to associate the pattern with an existing thumbnail, and this way you could also solve miscategorization. If the thumbnail doesn't exist they could be directed to wp.org thread and make a request for a new thumbnail/pattern category.

Here's a rough example with some really crappy thumbnails for another purpose: ![](https://user-images.githubusercontent.com/133038357/244879891-e07920c9-57eb-469f-a6f7-50214813dada.png)

**Comment by carolinan** [comment]
Thank you. @Ren2049. This is outside the scope of this discussion. The best place to put forward your suggestion is in the repository for the pattern directory: https://github.com/WordPress/pattern-directory/issues

**Comment by hanneslsm** [comment]
A few thoughts. I am not a big fan of the block pattern library. Reasons for that are:

**images**
- Images can be embedded in blocks that are hosted on a different server. Hello GDPR! ✌️
- When images are hosted on a different server, I cannot ensure reliability. It _could_ happen that someone who has control over the server could change all images with images of … [blob fishes ](https://www.google.com/search?q=blob+fish&oq=blob+fish&aqs=chrome..69i57.1925j0j1&sourceid=chrome&ie=UTF-8)
- What about liability for copyright infringement?

**design & presets**
I don't like the default presets and find them way too limiting. That's why I am creating a custom themes (wasn't this the idea of themes anyways?). Designers take a lot of time to think about choice of colors, typography, spacing, etc.. They are the ones who should create fitting patterns. If users start to use patterns that doesn't fit to the theme it will only make a lot of websites very ugly. Even if we had more "high quality" pattern, it doesn't matter how beautiful they look when they do not fit in context.

I feel like the pattern can hurt the reputation of Wordpress much more than it benefit from it.
I'd vote against including the pattern directory more into the editor.

**Reply by carolinan** [reply]
If person B wants to create something that person A think is ugly, it is in their right to do that, even if person A created the original theme.

**Reply by hanneslsm** [reply]
> If person B wants to create something that person A think is ugly, it is in their right to do that, even if person A created the original theme.

Absolutely, I fully agree! I'm also not arguing against the pattern directory in general, I only wanted to point out that I'm not in favour with the ideas listed above (embedding the pattern directly more into the editor), because of the said reasons. That's what this discussion was about, wasn't it?

**Comment by carolinan** [comment]
This is also out of the scope of this discussion. Please, for suggestions for improvements to the pattern directory, use https://github.com/WordPress/pattern-directory/issues

**Comment by carolinan** [comment]
Since these discussions are not productive here, and suggestions for improvement would not reach the right people who are actively working on the pattern directory, I will close this.

**Comment by kafleg** [comment]
I'm actively reviewing all the patterns that are submitted in the repository. If the patterns are too basic(like using 2 blocks without creativity) are unlisted. If the design is pretty unique, I approve them. 

We don't have guidelines for the design quality, but if the patterns are too basic and can easily create via few clicks, I don't approve them. 

But, we should allow everyone to contribute to the patterns directory.

**Comment by kafleg** [comment]
Also, I have an idea to make a patterns directory even better and more inclusive for everyone. I'll create meta ticket and can be discuss that things there.

**Reply by kafleg** [reply]
https://github.com/WordPress/pattern-directory/issues/599
